fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java

```diff
@@ -59,7 +61,9 @@
         @Override
         public String apply(final Event ev) {
             try {
-                final String id = ev.getIdentifier();
+                // build id from nodepath+user to collapse multiple nodes from adding/removing content nodes
+                final String id = ev.getPath().replaceAll("/jcr:content/","/").replaceAll(propPattern,"")
+                        + "-" + ev.getUserID();
```

Comment:
In doing some manual testing, it looks like NonRdfSources named "/book/ucsd:page" are being incorrectly sorted down to "/book".